basic/scraper.go

```diff
@@ -19,38 +19,44 @@ var (
 
 type Scraper struct {
 	// params
-	instance *config.Instance
-	exporter *Exporter
-	ch       chan<- prometheus.Metric
+	instance  *config.Instance
+	collector *Collector
+	ch        chan<- prometheus.Metric
 
 	// internal
-	svc    *cloudwatch.CloudWatch
-	labels []string
+	svc         *cloudwatch.CloudWatch
+	constLabels prometheus.Labels
 }
 
-func NewScraper(instance *config.Instance, exporter *Exporter, ch chan<- prometheus.Metric) *Scraper {
+func NewScraper(instance *config.Instance, collector *Collector, ch chan<- prometheus.Metric) *Scraper {
 	// Create CloudWatch client
-	sess, _ := exporter.sessions.GetSession(instance.Region, instance.Instance)
+	sess, _ := collector.sessions.GetSession(instance.Region, instance.Instance)
 	if sess == nil {
 		return nil
 	}
 	svc := cloudwatch.New(sess)
 
-	// Create labels for all metrics
-	labels := []string{
-		instance.Instance,
-		instance.Region,
+	constLabels := prometheus.Labels{
```

Comment:
constLabels? What's the difference?

Reply:
Constant labels are part of the metric's description; variable labels are not. The distinction is important for checked Collectors, but we switched to unchecked in this PR. Some more info: https://godoc.org/github.com/prometheus/client_golang/prometheus#hdr-Custom_Collectors_and_constant_Metrics